plugins/usi-dock/fu-usi-dock-mcu-device.c

```diff
@@ -718,12 +735,15 @@ fu_usi_dock_mcu_device_cleanup(FuDevice *device,
 	fwupd_request_add_flag(request, FWUPD_REQUEST_FLAG_ALLOW_GENERIC_MESSAGE);
 	fwupd_request_set_message(
 	    request,
-	    "The update will continue when the device USB cable has been re-inserted.");
+	    "Please unplug the USB cable. The update will continue in 30 seconds from now.");
 	fu_device_emit_request(device, request);
 
 	/* success */
 	fu_device_set_remove_delay(device, 900000);
 	fu_device_add_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
+
+	fu_progress_set_status(progress, FWUPD_STATUS_DEVICE_BUSY);
+	g_timeout_add_seconds(40, fu_usi_dock_mcu_device_insert_cb, device);
```

Comment:
this is much better, but I think you should do `g_timeout_add_seconds_full(....` with a `g_object_ref(device) so that the ref on FuDevice stays valid until the callback fires.